src/chains/ethereum/ethereum/src/transaction-pool.ts

```diff
@@ -185,15 +216,69 @@ export default class TransactionPool extends Emittery.Typed<{}, "drain"> {
         // nonce and mark as executable
         transactionNonce = transactorNonce ? transactorNonce : 0n;
         highestNonce = transactionNonce;
-        isExecutableTransaction = true;
+        transactionPlacement = TriageOptions.Executable;
         transaction.nonce = Quantity.from(transactionNonce);
       } else if (transactionNonce < transactorNonce) {
         // it's an error if the transaction's nonce is <= the persisted nonce
         throw new Error(
           `the tx doesn't have the correct nonce. account has nonce of: ${transactorNonce} tx has nonce of: ${transactionNonce}`
         );
       } else if (transactionNonce === transactorNonce) {
-        isExecutableTransaction = true;
+        transactionPlacement = TriageOptions.Executable;
+      }
+    }
+
+    // we have future transactions for this origin, this transaction is not yet
+    // executable, and this transaction is not replacing a previously queued/
+    // executable transaction, then this is potentially eligible to replace a
+    // future transaction
+    if (
+      queuedOriginTransactions &&
+      transactionPlacement !== TriageOptions.Executable &&
+      transactionPlacement !== TriageOptions.ReplacesPendingExecutable &&
+      (length = queuedOriginTransactions.length)
+    ) {
+      // check if a transaction with the same nonce is in the origin's
+      // future queue already. Replace the matching transaction or throw this
+      // new transaction away as necessary.
+      const queuedArray = queuedOriginTransactions.array;
+      const priceBump = this.#priceBump;
+      const newGasPrice = transaction.effectiveGasPrice.toBigInt();
+      // Notice: we're iterating over the raw heap array, which isn't
+      // necessarily sorted
+      for (let i = 0; i < length; i++) {
```

Comment:
This is mostly the same as the for loop [above](https://github.com/trufflesuite/ganache/pull/1237/files#diff-ca13abb534f8ddc94c9f9ae1309a903ad18f3f06f07b94653412d29f006178b2R151-R187), the differences being:
- above we set `transactionPlacement` to `TriageOptions.ReplacesExecutableTransaction`; here we set it to `TriageOptions.ReplacesFutureTransaction`
- above we need to track the `highestNonce`; here we do not.

I couldn't think of a great way to DRY this up. Every time I tried I only made it uglier. But I'm definitely open to suggestions.